libutils/string_lib.c

```diff
@@ -1801,3 +1801,27 @@ char *StringJoin(const Seq *const seq, const char *sep)
     char *const data = StringWriterClose(writer);
     return data;
 }
+
+Seq *StringSplit(const char *const str, const char *const del)
+{
+    assert(str != NULL);
+    assert(del != NULL);
+
+    Seq *const seq = SeqNew(1, free);
```

Comment:
IMHO, this `const` introduces more confusion than benefits.

Reply:
I don't agree, neither does [CONTRIBUTING.md](https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#safety--correctness)

Reply:
Well, that doesn't change the fact that for me it introduces an extra round of "OK, what does this `const` even mean. OK, it makes `seq` a constant variable. Which makes no difference in this code. Why is it there then?"

It's hard to be the only one with this opinion against two `const`-lovers, but I can't do no more.

Reply:
I think it's exactly the opposite - when something isn't const is when i have to pay extra attention to what it's value is and when it changes. Furthermore, this much fuzz over someone adding a const to their code is ridiculous.

Reply:
I'm sorry guys, but you are inventing a different language here. We haven't been (over)using `const` like this in the past. And generally this is not how C code is written in the vast majority of cases. I've been participating in a couple C projects and none of them used `const` like this. We are using quite a few C dependencies and none of them use `const` like this. Maybe it looks nice and feels natural to you, but this is not how it's normally done. So for deviating from the standard practices, we should have a really good reason.

Reply:
> Furthermore, this much fuzz over someone adding a `const` to their code is ridiculous.

Not really. Just the discussion probably belongs somewhere else.